src/vision/status_quo/alan_misses_c_sharp_async.md

```diff
@@ -0,0 +1,342 @@
+        source.Cancel();
+    });
+
+    await ExpensiveWork();
+}
+```
+
+Alan does some research and can't find anything similar. He reads that "dropping a future is
+cancelling it". In his junior dev days, Alan might have run with that idea and moved on to the next
+task, but experienced Alan knows something is wrong here. If he drops a `Future` how does he control
+the cleanup? Which `await` point is the one that will not be processed? This scares Alan since he
+realized he could get some really nasty bugs if this happens in production. In order to work around
+this, Alan needs to make sure *every* future around critical code is carefully reviewed for drops in
+the wrong places. Alan also decided he needs to come up with some custom code to handle cancelling.
+
+// Do we need an example impl of a rust version of cancellation?
+
+## 🤔 Frequently Asked Questions
+
+### **What are the morals of the story?**
+* A lot of async utilities that are common in the standard library for other languages only exists
+in third-party libraries or doesn't exist at all.
+* It's not obvious that futures could only run part-way.
+
+### **What are the sources for this story?**
+* The docs for [`oneshot::Canceled`] mentions that dropping a `Sender` will cancel the future.
+Someone inexperienced might accidentally apply this to a broader scope of types.
+* [This IRLO post] has a nice discussion on cancellation, where the [linked gist]
+is a thorough overview of problems surrounding cancelation in async rust, with comparisons to other
+languages.
+
+[`oneshot::Canceled`]: https://docs.rs/futures/0.3.15/futures/channel/oneshot/struct.Canceled.html
+[This IRLO post]: https://internals.rust-lang.org/t/async-await-the-challenges-besides-syntax-cancellation/10287
+[linked gist]: https://gist.github.com/Matthias247/ffc0f189742abf6aa41a226fe07398a8
+
+### **Why did you choose *NAME* to tell this story?**
+C# is a garbage collected language that has had async for a long time. Alan best fit the model for
+a developer coming from such a language.
+
+### **How would this story have played out differently for the other characters?**
+* [Barbara] may already be used to the ideosynchracies of async in rust. She may not realize how
+difficult it could be for someone who has a very different model of async engrained into them.
+* [Grace] has likely never used async utilities similar to the ones in C# and other GC languages. C
+and C++ tend to use callbacks to manage async workflows. She may have been following the C++
+proposals for coroutines (e.g. `co_await`, `co_yield`, `co_return`), but similar to rust, the
+utilities are not yet thoroughly built out in those spaces. She may be familiar with cancelation in
+external libraries like [`cppcoro`](https://github.com/lewissbaker/cppcoro#Cancellation), or async in
+general with [`continuable`](https://github.com/Naios/continuable)
+* [Niklaus] may not have had enough experience to be wary of some of the pitfalls encountered here.
```

Comment:
I also wasn't sure if this was helpful to have here. I wouldn't expect someone new to programming to be familiar with these pitfalls, but is this too negative of a sentiment to have here?